### PR TITLE
Addition of the Rust Programming Language to SHELTER-ZONE Coder_Cert.

### PR DESCRIPTION
**Pull request by HTGAzureX1212**
This Pull Request adds a new folder, `Rust_Cert`, and includes `Rust_Cert.rs` - the Rust script itself and a `Cargo.toml` file, which is required dependencies for the Rust script.

### CODE REVIEW

File: Rust_Cert/Rust_Cert.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+impl<'a> LockSolver<'a> {
+    fn new<F>(solve_method: &'a F) -> LockSolver<'a> where F: Fn(&String) -> String {
+        return LockSolver {
+            test_answers: vec![],
+            solve_method,
+            test_cases: from_str(&*initialize()).unwrap(),
+        }
+    }
+
+    fn run(self_: &mut Self) {
```
Comment (HTGAzureX1212) [inline Rust_Cert/Rust_Cert.rs:42]: A little mutation here:
```rs
fn run(&self) {
```

File: Rust_Cert/Rust_Cert.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        let method: &dyn Fn(&String) -> String = self_.solve_method;
```
Comment (HTGAzureX1212) [inline Rust_Cert/Rust_Cert.rs:43]: To comply with the change at the definition of the `run` method:
```rs
let method: &dyn Fn(&String) -> String = self.solve_method;
```

File: Rust_Cert/Rust_Cert.rs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        let sample_test = method(&String::from("12345"));
+        let mut lock_number: i32 = 1;
+
+        if sample_test != "2353" {
+            panic!(format!("Your `solve` method returned {} instead of \"2353\"", sample_test));
+        }
+
+        for (key, value) in self_.test_cases.iter() {
```
Comment (HTGAzureX1212) [inline Rust_Cert/Rust_Cert.rs:51]: To comply with the change at the definition of the `run` method:
```rs
for (key, value) in self.test_cases.iter() {
```

File: Rust_Cert/Rust_Cert.rs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            if &method_result != value && value != "" {
+                panic!(format!("Failed to open lock #{}, your solution is probably wrong - please \
+                make sure that your `solve` method is correct.", lock_number));
+            }
+
+            lock_number += 1;
+        }
+    }
+
+    fn get_password(self_: &Self, user_id: String) -> String {
```
Comment (HTGAzureX1212) [inline Rust_Cert/Rust_Cert.rs:63]: Same mutation as the definition of `run` method:
```rs
fn get_password(&self, user_id: String) -> String {
```

File: Rust_Cert/Rust_Cert.rs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        let mut sha1_hasher: Sha1 = Sha1::new();
+        let mut answer: String = self_.test_answers.join("");
```
Comment (HTGAzureX1212) [inline Rust_Cert/Rust_Cert.rs:65]: To comply with the change at the definition of the `get_password` method:
```rs
let mut answer: String = self.test_answers.join("");
```

**Review by HTGAzureX1212 — COMMENTED** [review]
Performing 5 mutations to further simplify the definition of arguments.

**Comment by HTGAzureX1212** [comment]
Just a follow-up: since `class`es do __NOT__ exist in the Rust language, a `struct`-and-`impl` combination is used to simulate the behaviour of `class` in languages where `class`es exist.

**Comment by Proladon** [comment]
感謝貢獻，但基於以下原因我將要把此PR暫時吊掛
SZ裡使用RUST的人數過少，也沒人在討論關於RUST的話題
如果要新增RUST還要新增對應的身分組，對於目前來說會增加管理難度且非常的沒有必要性
因此當SZ內需要RUST時，才會再次重新審視此PR
但對於社群的任何的貢獻都值得鼓勵!

**Pull request closed without merging**